src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -1800,18 +1808,19 @@ SortedShardIntervalArray(List *shardIntervalList)
 
 	/* sort shard intervals by their minimum values in ascending order */
 	qsort_arg(shardIntervalArray, shardIntervalCount, sizeof(ShardInterval *),
-			  (qsort_arg_comparator) CompareShardIntervals, (void *) typeCompareFunction);
+			  (qsort_arg_comparator) CompareShardIntervalPointers,
+			  (void *) typeCompareFunction);
 
 	return shardIntervalArray;
 }
 
 
 /*
- * CompareShardIntervals acts as a helper function to compare two shard interval
+ * CompareShardIntervalPointers acts as a helper function to compare two shard interval
  * pointers by their minimum values, using the value's type comparison function.
  */
 static int
-CompareShardIntervals(const void *leftElement, const void *rightElement,
+CompareShardIntervalPointers(const void *leftElement, const void *rightElement,
 					  FmgrInfo *typeCompareFunction)
 {
 	ShardInterval **leftShardInterval = (ShardInterval **) leftElement;
```

Comment:
@marcocitus We'll be using a single CompareShardIntervals function in the branch we merge to the master. We haven't changed it since it is the patch that we've tested well.